tests/models/udop/test_processor_udop.py

```diff
@@ -22,9 +20,10 @@
 import numpy as np
 
 from transformers import (
-    PreTrainedTokenizer,
+    AutoProcessor,
     PreTrainedTokenizerBase,
     PreTrainedTokenizerFast,
+    UdopProcessor,
```

Comment:
UdpoProcessor is imported below if pytesseract is available, so imo we don't need to add it here

Reply:
I removed it from the pytesseract check instead, as there is a strange bug where the line below the class definition (`processor_class = UdopProcessor`) will still be executed even if the "requires" are not satisfied, which makes the CI break

Reply:
Does that mean UDOP has no dependency on pytesseract to run the processor test?

Reply:
Only the image_processor (`LayoutLMv3ImageProcessor`) depends on pytesseract, but since the import check is already done at the level of `LayoutLMv3ImageProcessor`, it doesn't seem to me that it should also be done when importing the processor? Though I'm not sure how these nested requirement checks should be dealt with.

Reply:
i see, that's weird because the Tester class has a `require_pytesseract` dependency which afaik is same as `is_pytesseract_available()`

Actually only importing processor has no dependencies, from what I see it doesn't use pytesseract directly. So it should be ok to import it as is, and the tests should be skipped by `require_pytesseract` if package is not installed. I'm just curious why that broke, if you have bandwidth to explore that. I couldn't reproduce it by removing `UdopProcessor` from general imports